### ISSUE TITLE
Limit retained history

### ISSUE THREAD

**Issue by reallyBasic** [r10s/messenger-android#22]
Similar to [#21](https://github.com/r10s/messenger-android/issues/21). I would like to have an option to limit the amount of history kept in the app as I'm using the app. For example, I could limit the amount of history by number of message (like 200) or length of time (like 90 days).

I would not expect this limit to cause messages to be deleted from the server.

**Comment by r10s** [comment]
This is planned. In fact, we already have an option for it (currently hidden at "System->Delta Chat App-info / Memory / Manage memory")

**Comment by r10s** [comment]
if we implement this feature, we should have different timeouts until messages from the "Contact requests" and messages from other chats are deleted (deleted from the device, _not_ from the server), this would target #132 then.

**Comment by csb0730** [comment]
> In fact, we already have an option for it (currently hidden at "System->Delta Chat App-info / Memory / Manage memory")

At my device I see here only "System->Delta Chat App-info / Memory / Delete memory" ??
No possibility to manage something.

Waiting urgently for an automatic feature ;-)

**Comment by r10s** [comment]
well, the ui has changed the last months :)

**Comment by csb0730** [comment]
My proposal for a *handy* function:
1) make messages stickable or pinable first (I don't know the best wording ;).
That means that a message can be marked manually and then it is not touched by an automatic garbage collection. This enables the user to keep important messages!
2) introduce an option to set max number of messages in a chat's setting (and show somewhere current number of messages in a chat).
3) trigger the garbage collection whether manually or automatic
(with new message in a chat or once per day or even with a button?)

Comments welcome ;-)

**Comment by csb0730** [comment]
Maybe this issue is better moved to the core?

**Comment by r10s** [comment]
yes, however, issues cannot be moved on github. so, i prefer most times to leave things unsorted over having fragmentation :)

**Comment by csb0730** [comment]
EDIT proposal for a handy function:

1. IMPORTANT: use starring of messages as it is foreseen yet.
That means that a message which is starred is not touched by an automatic garbage collection. This enables the user to keep important messages!
2. Introduce an option to set max number of messages in a chat's setting (and show somewhere current number of messages in a chat).
3. Trigger the garbage collection whether manually or automatic.
(with new message arriving in chat or per interval or even with a button to do manually?)

**Comment by testbird** [comment]
The last comment looks identical to what you wrote three messages ago, and what you posted at https://github.com/deltachat/deltachat-core/issues/120 [consolidated] (auto-)archive / delete (support continuous daily usage)

And UI point in https://github.com/deltachat/deltachat-core/issues/120 is:
> - [ ] UI feature to auto-archive older messages from chats (default those exceeding a 100 messages count?)

**Comment by csb0730** [comment]
With my own experience now while updating DC and the need of using the backup function for that, I've been running nearly into a memory constraint and reimporting data has been at risk not to finalize any more.
I'm using DC for a year now and database grows and grows. 1GB (database file only) is reached meanwhile.
With this comment I want to remind that this issue here is important because there is no possiblity to shrink the database file by deleting messages. Especially standard users have no chance to do anything except new installation of DC with an empty new database!

The database compression (Point 4. / vacuum function) should be attacked as soon as possible!

And finally as a 5th point I want to propose an archive funktion to a(some) second database file(s) which could be opened from DC to look for older archieved messages and/or copied from the device to another storage location.

**Comment by csb0730** [comment]
6. it should be possible to define the external SD storage card for attachments.

**Comment by testbird** [comment]
Maybe open different issues for 4. and 6.?

The other points are part of https://github.com/deltachat/deltachat-core/issues/120
Should I also add 5. there, so all (auto) archiving and deleting stuff is kept together, and this issue can be closed?

**Comment by testbird** [comment]
ok, added 5. to deltachat/deltachat-core#120

**Comment by csb0730** [comment]
Point 4/5/6 are the technical functions which complete point 1-3 (user functions). Only with 4-6 the other functions are really fully useful (!)

And yes, maybe it is worth to move them to a separate issue. Will do that later ;-)

**Comment by csb0730** [comment]
As I saw @r10s wants to keep this issue open to handle all this stuff. And therefore I included all ideas to that here ;-)

**Comment by testbird** [comment]
I don't know, the issues are clearly unsorted *and* fragmented, and thus there is no clear path, and streamlined progress possible on the issues (with limited on/off time constrains). Quickly solvable topics could be lined up better, when things are sorted out.
